std/os/index.zig

```diff
@@ -710,13 +710,15 @@ pub fn getEnvMap(allocator: *Allocator) !BufMap {
             const key_start = i;
 
             while (ptr[i] != 0 and ptr[i] != '=') : (i += 1) {}
-            const key = ptr[key_start..i];
+            const key = try std.unicode.utf16leToUtf8Alloc(allocator, ptr[key_start..i]);
```

Comment:
This code looks correct, but I think it can be more optimal. If you use the iterator API here you can avoid heap allocation. For example:

https://github.com/ziglang/zig/blob/b953e322a857e58df2209ca7ae9e965b9e768887/std/unicode.zig#L500-L505

You could use `Utf16LeIterator` instead of `utf16leToUtf8Alloc`.

Same with the `value` below.

Reply:
Added